pkg/policy/rule_test.go

```diff
@@ -1654,3 +1646,105 @@ func (ds *PolicyTestSuite) TestEgressL4AllowAll(c *C) {
 	// must have empty endpoint selector to match on all sources
 	c.Assert(len(filter.Endpoints), Equals, 0)
 }
+
+func (ds *PolicyTestSuite) TestL4WildcardMerge(c *C) {
+
+	// Test the case where if we have rules that select the same endpoint on the
+	// same port-protocol tuple with one that is L4-only, and the other applying
+	// at L4 and L7, that the L4-only rule shadows the L4-L7 rule. This is because
+	// L4-only rule implicitly allows all traffic at L7, so the L7-related
+	// parts of the L4-L7 rule are useless.
+	repo := parseAndAddRules(c, api.Rules{&api.Rule{
+		EndpointSelector: endpointSelectorA,
+		Ingress: []api.IngressRule{
+			{
+				FromEndpoints: []api.EndpointSelector{endpointSelectorC},
+				ToPorts: []api.PortRule{{
+					Ports: []api.PortProtocol{
+						{Port: "80", Protocol: api.ProtoTCP},
+					},
+					Rules: &api.L7Rules{
+						HTTP: []api.PortRuleHTTP{
+							{Method: "GET", Path: "/"},
+						},
+					},
+				}},
+			},
+			{
+				ToPorts: []api.PortRule{{
+					Ports: []api.PortProtocol{
+						{Port: "80", Protocol: api.ProtoTCP},
+					},
+				}},
+			},
+		},
+	}})
+
+	buffer := new(bytes.Buffer)
+	ctx := SearchContext{To: labelsA, Trace: TRACE_VERBOSE}
+	ctx.Logging = logging.NewLogBackend(buffer, "", 0)
+
+	l4IngressPolicy, err := repo.ResolveL4IngressPolicy(&ctx)
+	c.Assert(err, IsNil)
+
+	c.Log(buffer)
+
+	filter, ok := (*l4IngressPolicy)["80/TCP"]
+	c.Assert(ok, Equals, true)
+	c.Assert(filter.Port, Equals, 80)
+	c.Assert(filter.Ingress, Equals, true)
+
+	c.Assert(len(filter.Endpoints), Equals, 0)
+
+	c.Assert(filter.L7Parser, Equals, L7ParserType(""))
+	c.Assert(len(filter.L7RulesPerEp), Equals, 0)
+
```

Comment:
Please add asserts on `filter.DerivedFromRules`. I'm curious about its contents here.
IMHO it should only refer to the L4-only rule in this case.

Reply:
I'll punt this for further discussion based off of my comments above. Let's discuss this and nail down what exactly `DerivedFromRules` means.